### ISSUE TITLE
SQLite3Exception: no such function: ST_GeomFromWKB

### ISSUE THREAD

**Issue by jrmgx** [brick/geo#52]
Hello, I'm following the doc, and I'm using the SQLite engine (I was using GEOSEngine engine, but it does not have split, that's a bit sad)

Anyway, I've installed libsqlite3-mod-spatialite and I set up the SQLite3Engine like this

```php 
$sqlite3 = new SQLite3(':memory:');
$sqlite3->loadExtension('mod_spatialite.so');
$sqlite3->query('SELECT InitSpatialMetaData();');
$this->geometryEngine = new SQLite3Engine($sqlite3);
```

but when I do any kind of geo work I have exceptions like those:
`GeometryEngineException: This operation is not supported by the geometry engine.`
and down the stack:
`SQLite3Exception: no such function: ST_GeomFromWKB`

So it says _This operation is not supported by the geometry engine_, but I'm pretty much sure it is.

This code (except split) was working nicely on GEOSEngine, so I'm quite sure it's not about the implementation, but more about the SQLite initialization.

Any idea what I'm doing wrong?

**Comment by jrmgx** [comment]
My bad, the `loadExtension` and `query` return `false` I thought it would throw exceptions.
The mod_spatialite.so was hidden in `/usr/lib/x86_64-linux-gnu/` (Ubuntu)

**Comment by BenMorel** [comment]
No problem, actually I think this error message is a bit misleading, maybe it should just be:

> The geometry engine returned an error: {message}

and include the original exception message, to avoid having to dig the stack to find the source of the problem.

PR welcome!

**Comment by BenMorel** [comment]
Geometry engine exception messages have been improved in 5770b9048f51c6854926557e0d0b422d5b672eab and released in version [0.12.0](https://github.com/brick/geo/releases/tag/0.12.0).
